### ISSUE TITLE
How to use non registered toolbox 

### ISSUE THREAD

**Issue by ragavsathish** [ToolboxHub/ToolboxToolbox#61]
Hi, 
Thanks for the wonderful dependency management tool. I was successfully able to deploy the toolbox, but when I say tbUse() toolbox is not getting added in the Matlab path.

Kindly clarify
```
>> record = tbToolboxRecord('name', 'mdepin', 'type', 'git', 'url', 'https://github.com/ragavsathish/mdepin.git');
configPath = fullfile(tempdir(), 'mdepin-config.json');
tbWriteConfig(record, 'configPath', configPath);
tbDeployToolboxes('configPath', configPath)
Updating "ToolboxRegistry".
fatal: Not a git repository (or any of the parent directories): .git
Unable to update toolbox registry "ToolboxRegistry", proceeding with current version.
Obtaining "mdepin".
Cloning into 'D:\project\repo\toolboxes\mdepin'...
Adding "ToolboxToolbox" to path at "D:\project\repo\ToolboxToolbox".
Adding "mdepin" to path at "D:\project\repo\toolboxes\mdepin".
Checking for "mdepin" local hook.
Looks good: 1 resolved toolboxes deployed OK.

ans = 

                extra: [1x0 char]
               flavor: [1x0 char]
                 hook: [1x0 char]
           importance: [1x0 char]
    localHookTemplate: [1x0 char]
                 name: 'mdepin'
        pathPlacement: 'append'
      requirementHook: [1x0 char]
            subfolder: [1x0 char]
          toolboxRoot: [1x0 char]
                 type: 'git'
               update: [1x0 char]
                  url: 'https://github.com/ragavsathish/mdepin.git'
                 path: ''
               status: 0
              message: 'Cloning into 'D:\project\repo\toolboxes\mdepin'...'
            operation: 'obtain'
              command: 'git clone "https://github.com/ragavsathish/mdepin.git" "D:\project\repo\toolboxes\mdepin"'
             strategy: [1x1 TbGitStrategy]
                 isOk: 1

>> tbUse('mdepin')
Updating "ToolboxRegistry".
fatal: Not a git repository (or any of the parent directories): .git
Unable to update toolbox registry "ToolboxRegistry", proceeding with current version.
Unable to resolve any configurations.
  configPath to try loading was: D:\project\repo\toolbox_config.json
  explicit config struct contained 1 records.
  registered toolboxes had names: "mdepin", .
Proceeding in case there's a hook or localHook to be run.
Adding "ToolboxToolbox" to path at "D:\project\repo\ToolboxToolbox".
Checking for "mdepin" local hook.

ans = 

0x0 struct array with fields:

    extra
    flavor
    hook
    importance
    localHookTemplate
    name
    pathPlacement
    requirementHook
    subfolder
    toolboxRoot
    type
    update
    url
```

**Comment by DavidBrainard** [comment]
Hi,

a) Let's first test that your installation works as expected for a toolbox that we know works.

Try tbUse('BrainardLabToolbox').  This should cause the TbTb to go out, grab this toolbox from gitHub, put it into your toolboxes folder, and put it on your Matlab path.  You an tell if it worked if you execute
FWHMToStd(10) and get 4.2466... as the answer.

If this works, you can then go delete the BrainardLabToolbox tree from your toolboxes folder.  If it doesn't, something isn't quite right about your ToolboxToolbox configuration.

You are the first person as far as we know to try this on Windows, so it could be that there is a portability issue you are encountering.  The particular thing that doesn't look right is that your system is failing
when it tries to update the toolbox registry.

b) I take it from your message that your tbDeployToolboxes code is working as you expect, other than the fact that the ToolboxRegistry update isn't working. And that won't break anything else as long as you don't need anything updated from the registry.  If so, then I think the issue with tbUse('mdepin') is that you need to make a copy of the mdepin-config.json file you created in your tempdir and put it into ToolboxRegistry/configurations/mdepin.json.  (Note, no "-config" in the filename.)  tbUse works by looking in the registry for the appropriate .json file, which it deduces from the string passed to tbUse, and going from there.

c) If you get this to work, we're happy to accept pull requests on ToolboxRegistry that add your .json files to the registry.  Once it is set up, tbUse should then know what to do with tbUse('mdepin') from any machine that has ToolboxToolbox installed, without need to do anything additional (although the git command is going to have to be able to install/update the toolbox registry - you shouldn't need to do that by hand, not even the first time.)

Let us know how this goes.

Best,

David Brainard

**Comment by ragavsathish** [comment]
Hi Professor Brainard,

Thanks for the detailed notes.

a)  There was a small bug in building windows command for git updates. I sent the pull request for the same. If it is convincing, Kindly merge.
b) As you advised, renaming ` mdepin-config.json` to `mdepin.json` and dropping it inside the ToolboxRegistry/configurations/ made it work. 

Thanks again,
Sathish

**Comment by DavidBrainard** [comment]
Great.  Ben Heasly, who developed the TbTb will review your PR next week.

If you'd like your mdepin.json in the ToolboxRegistry, just do a pull request for it.

Best,

David

**Comment by benjamin-heasly** [comment]
Merged #62 to close this issue.  Thanks!  Please stay in touch if you have other issues.
